freqtrade/optimize/backtesting.py

```diff
@@ -479,6 +472,13 @@ def _enter_trade(self, pair: str, row: Tuple, stake_amount: Optional[float] = No
         min_stake_amount = self.exchange.get_min_pair_stake_amount(pair, propose_rate, -0.05) or 0
         max_stake_amount = self.wallets.get_available_stake_amount()
 
+        pos_adjust = trade is not None
+        if stake_amount is None:
```

Comment:
i think you missunderstood this slightly.

In my opinion, you can remove the `if stake_amount is None` condition - and move it into the subsequent IF - as it's either `stake_amount is None and not pos_adjust` ... but never only one OR the other.

Reply:
I agree with them being currently the same since stake_amount and trade either are sent together or not BUT I'd leave them like this for now since that might not always be the case in the future and this way it's more "safe".

Reply:
Fine, it actually does look more readable like that.